### ISSUE TITLE
Canonical URLs overhaul

### ISSUE THREAD

**Issue by michaeltorbert** [awesomemotive/all-in-one-seo-pack#1631]
We have various issues open for [canonical urls](https://github.com/semperfiwebdesign/all-in-one-seo-pack/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+canonical).

When we first added the feature, WordPress didn't actually have a canonical url. I *think all pages should have one now.
We should probably consider just leaving it alone unless specifically changed. Let's investigate what to do about this. 

Also, when we _are_ changing it, we may want to consider using built-in functions/hooks that now exist. For example, the get_canonical_url filter (WP 4.6), wp_get_canonical_url() (WP 4.6), etc.

-----

(Copied from duplicate issue)

We should probably rethink what we want to do with canonical URLs. When we first introduced this, WordPress didn't have canonical URLs functionality. Now they do.
My suggestion is that we just show the already existing WP canonical URL, and only change it if the user explicitly wants a custom one.

Suggested Workflow:
We access the WP canonical URL via WP api, and display that in our section, then have it filterable as needed, either internally or by users.

In other words, much/most of the canonical URL code can be removed, and replaced with a bit of new code in it's own file.
